File: Theory of formal languages/parser/Application.cpp
#include "modules/Application.h"
#include "io/FileReadUtils.h"
#include <iostream>

Application::Application(const std::string &codeFileName, const std::string &grammarFileName) : codeFileName(codeFileName),grammarFileName(grammarFileName) {}

void Application::start() {
    std::string s=readFile(codeFileName);
    Lexer lexer;
    std::string translatedCode=lexer.translate(s);
    std::cout<<translatedCode<<std::endl;
    Parser parser=Parser(translatedCode,grammarFileName);
    if (parser.parse()==Parser::SUCCESS){
        std::cout<<"Wonderful"<<std::endl;
    }else{
        std::cout<<":("<<std::endl;
    }
}

File: Compilers/lab7.1/src/bmstu/iu9/Application.java
package bmstu.iu9;

import bmstu.iu9.parser.ParseStatus;
import bmstu.iu9.grammar.Grammar;
import bmstu.iu9.io.FileUtils;
import bmstu.iu9.parser.AbstractParser;
import bmstu.iu9.scanner.AbstractScanner;
import bmstu.iu9.modules.ICompiler;
import bmstu.iu9.modules.Compiler;
import bmstu.iu9.parser.ParseResult;
import bmstu.iu9.tokens.EndToken;
import bmstu.iu9.tokens.IToken;

import java.util.ArrayList;
import java.util.List;

public final class Application {
    public void start(String fileName,String grammarFileName){
        String program= FileUtils.readFile(fileName);
        ICompiler compiler = new Compiler();
        AbstractScanner scanner=compiler.getScanner(program);
        List<IToken> tokenStream = new ArrayList<>();

        for (IToken token = scanner.getNextToken(); token!= EndToken.getInstance(); token=scanner.getNextToken()){
            tokenStream.add(token);
            //System.out.println(token);
        }
        tokenStream.add(EndToken.getInstance());

        AbstractParser parser = compiler.getParser(new Grammar(grammarFileName),tokenStream);
        ParseResult result = parser.parse();
        if (result.getStatus() == ParseStatus.SUCCESS){
            System.out.println(result.getSyntaxTree());
        }
        System.out.println(result.getMessage());
    }
}
